projects/user_publication/deduplicate.py

```diff
@@ -83,38 +83,37 @@ def flag_duplicate_in_whole_db():
         if pub in duplicates:
             with transaction.atomic():
                 pub.status = Publication.STATUS_DUPLICATE
-                pub.reviewed = False
+                pub.checked_for_duplicates = False
                 pub.save()
 
 
 def pick_duplicate_from_pubs(dpub, opub):
+    # decide which publication should be categorized as duplicate
     # this logic can further extend, currently its for if publication is preprint
-    preprint_in_duplicate = (
-        'preprint' in dpub.forum if dpub.forum else ""
-        or 'preprint' in dpub.bibtex_source
-        or 'arXiv' in dpub.doi if dpub.doi else ""
-        or 'preprint' in dpub.publication_type
-    )
-    preprint_in_original = (
-        'preprint' in opub.forum if opub.forum else ""
-        or 'preprint' in opub.bibtex_source
-        or 'arXiv' in opub.doi if opub.doi else ""
-        or 'preprint' in opub.publication_type
-    )
-    if preprint_in_duplicate or preprint_in_original:
-        # if preprint is both in duplicate and original pick duplicate as duplicate
-        if preprint_in_duplicate and preprint_in_original:
-            duplicate_pub = dpub
-            original_pub = opub
-        elif preprint_in_duplicate:
-            duplicate_pub = dpub
-            original_pub = opub
-        else:
-            duplicate_pub = opub
-            original_pub = dpub
-    else:
+    def has_preprint(publication):
+        return (
+            "preprint" in publication.forum.lower()
+            if publication.forum
+            else ""
+            or "preprint" in publication.bibtex_source.lower()
+            or "arxiv" in publication.doi.lower()
+            if publication.doi
+            else "" or "preprint" in publication.publication_type.lower()
+        )
+
+    duplicate_has_preprint = has_preprint(dpub)
+    original_has_preprint = has_preprint(opub)
+
+    if duplicate_has_preprint and original_has_preprint:
+        duplicate_pub = dpub
+        original_pub = opub
+    elif duplicate_has_preprint:
         duplicate_pub = dpub
         original_pub = opub
```

Comment:
I'm confused by this. Both of these branches have the same body. Could we not just check 
```python
if duplicate_has_preprint:
    duplicate_pub = dpub
    original_pub = opub
else:
    duplicate_pub = opub
    original_pub = dpub
```